### PR TITLE
test: e2e tests for contentEditable input bar and paste tiles

### PR DESCRIPTION
**Pull request by wenxi-onyx**
## Description

Playwright e2e tests for the contentEditable input bar and paste tile feature (#10810).

Also includes a minor production change: added `data-testid="paste-tile-backdrop"` to the popover backdrop in `PasteTilePopover.tsx` for stable test targeting.

**Test suites (61 tests across 12 describe blocks):**
- **Core Text Input & Submission** (6) — typing, Enter/click send, empty/spaces input guards, input cleared after send, long messages
- **Multiline Input** (3) — Shift+Enter newlines, height growth, newline preservation
- **Paste Behavior** (3) — plain text, HTML stripping, select-all replace
- **Paste Security** (4) — XSS via script/img/event handlers/rich HTML
- **Auto-Resize** (4) — grows, shrinks, max height cap, scrollable overflow
- **Placeholder** (4) — shows on load, hides on type, restores on delete/send
- **Focus Management** (3) — focused on load, re-focused after send, click away/back
- **Prompt Shortcuts** (1) — typing / triggers UI
- **Keyboard Edge Cases** (4) — backspace, Ctrl+A+backspace, Ctrl+A+type, inline spans no spurious newlines
- **Paste Tiles** (24) — tile creation/removal, preview/meta, small text bypass, submit with tile, popover open/edit/dismiss, keyboard nav (arrow highlight, Enter opens popover, typing deselects, second arrow moves past), Backspace highlight+delete, Ctrl+A/C/X, multiple tiles, cursor hidden on highlight, clear-to-empty removes tile, edit-then-send verifies synced state, text before+after tile preserved, backdrop click dismisses popover
- **Paste Tiles — User Setting** (2) — disabled by default, enabled when toggled
- **Visual Regression** (3) — empty, with text, multiline screenshots

## How Has This Been Tested?

These ARE the tests. They run against a live Onyx deployment with mocked chat endpoints.

## Additional Options

- [ ] [Optional] Please cherry-pick this PR to the latest release version.
- [x] [Optional] Override Linear Check

### CODE REVIEW

**Comment by wenxi-onyx** [comment]
@greptileai @cubic-dev-ai complete re-review

File: web/tests/e2e/chat/input_bar_behaviors.spec.ts [hunk 1]
```diff
@@ -634,6 +564,376 @@ test.describe("Keyboard Edge Cases", () => {
...
+  test.beforeEach(async ({ page }, testInfo) => {
+    resetTurnCounter();
+    await page.context().clearCookies();
+    await loginAsWorkerUser(page, testInfo.workerIndex);
+    await page.goto("/app");
+    await page.waitForLoadState("networkidle");
+    // Enable paste tiles for this test suite
+    await page.evaluate(() =>
+      fetch("/api/paste-as-tile?paste_as_tile=true", { method: "PATCH" })
+    );
```
Comment (nmgarza5) [inline web/tests/e2e/chat/input_bar_behaviors.spec.ts]: should be added to https://github.com/onyx-dot-app/onyx/blob/2b30afc68f14b15e55ec301984c68f36fa6b5313/web/tests/e2e/utils/onyxApiClient.ts#L81

**Review by nmgarza5 — COMMENTED** [review]
Can you refactor the tests to use the ChatPage? https://github.com/onyx-dot-app/onyx/blob/2b30afc68f14b15e55ec301984c68f36fa6b5313/web/tests/e2e/chat/ChatPage.ts#L11


ideally, we clean up these tests quite a bit by adding more methods to ChatPage.ts similar to GroupsAdminPage https://github.com/onyx-dot-app/onyx/blob/2b30afc68f14b15e55ec301984c68f36fa6b5313/web/tests/e2e/admin/groups/GroupsAdminPage.ts#L12

example: https://github.com/onyx-dot-app/onyx/blob/7008b3b6c6932b0406227705c6e667cdaa0e5915/web/tests/e2e/admin/groups/groups.spec.ts#L98

**Comment by wenxi-onyx** [comment]
@greptile-ai re-review

File: web/tests/e2e/chat/ChatPage.ts [hunk 2]
```diff
@@ -77,4 +74,16 @@ export class ChatPage {
...
+  // Message assertions
+  // ---------------------------------------------------------------------------
+
+  async expectHumanMessage(text: string, index = 0): Promise<void> {
+    await expect(this.humanMessage(index)).toContainText(text);
+  }
+
+  async expectNoHumanMessages(): Promise<void> {
+    await expect(this.humanMessages).toHaveCount(0);
+  }
```
Comment (nmgarza5) [inline web/tests/e2e/chat/ChatPage.ts:88]: nit: might want to have some timeouts or explicit waits here to prevent flakiness. just a thought

**Review by nmgarza5 — APPROVED** [review]

File: web/tests/e2e/chat/ChatPage.ts (hunk 2 above)
Comment (wenxi-onyx) [reply web/tests/e2e/chat/ChatPage.ts:88]: toHaveCount idiomatically retries and has timeouts
